### PR TITLE
Add patch notes to versioning doc

### PR DESCRIPTION
**Pull request by mbforbes**

### CODE REVIEW

**Comment by mbforbes** [comment]
@zmerlynn can I haz quick review? (sorry for spamming you with reviews today)

File: docs/design/versioning.md [hunk 1]
```diff
@@ -68,6 +68,12 @@ Here is an example major release cycle:
 
 It may seem a bit strange to complete the v2 API before v2.0 is released, but *adding* a v2 API is not a breaking change. *Removing* the v2beta\* APIs *is* a breaking change, which is what necessitates the major version bump. There are other ways to do this, but having the major release be the fresh start of that release's API without the baggage of its beta versions seems most intuitive out of the available options.
 
+# Patches
+
+GKE only allows users to run at the latest path (x.x.**X**) of any given minor release (x.**X**.x). From a customer perspective, this means there should be no reason a customer would ever want to run a non-latest patch, as it will be impossible (via GKE).
```
Comment (zmerlynn) [inline docs/design/versioning.md]: Can we make this not GKE specific?

cc @bgrant0607 

(Nit: I assume you meant `latest patch`?)

Comment (mbforbes) [inline docs/design/versioning.md]: I was going to write "hosted providers such as GKE" but GKE's the only hosted provider I know that does this. Can you think of another wording?

And oops, yes, that's a bug. I'll push another commit.

File: docs/design/versioning.md [hunk 2]
```diff
@@ -68,6 +68,12 @@ Here is an example major release cycle:
 
 It may seem a bit strange to complete the v2 API before v2.0 is released, but *adding* a v2 API is not a breaking change. *Removing* the v2beta\* APIs *is* a breaking change, which is what necessitates the major version bump. There are other ways to do this, but having the major release be the fresh start of that release's API without the baggage of its beta versions seems most intuitive out of the available options.
 
+# Patches
+
+Google Container Engine only allows users to run at the latest patch (x.x.**X**) of any given minor release (x.**X**.x). From a customer perspective, this means there should be no reason a customer would ever want to run a non-latest patch, as it will be impossible (via Google Container Engine).
```
Comment (bgrant0607) [inline docs/design/versioning.md]: This paragraph seems entirely unnecessary.

File: docs/design/versioning.md [hunk 3]
```diff
@@ -68,6 +68,12 @@ Here is an example major release cycle:
...
+
+This critically trusts our patching process to be only for issues such as bugfixes and security vulnerabilities. More specifically, there should be no feature incompatibilities between patch versions. This includes dependency changes, such as modifying the Docker version.
```
Comment (bgrant0607) [inline docs/design/versioning.md]: See also: https://github.com/kubernetes/kubernetes/pull/12734#issuecomment-132697357

Proposed text:

Patch releases are intended for critical bug fixes to the latest minor version, such as addressing security vulnerabilities, fixes to problems affecting a large number of users, severe problems with no workaround, and blockers for products based on Kubernetes. 

They should not contain miscellaneous feature additions or improvements, and especially no incompatibilities should be introduced between patch versions of the same minor version (or even major version). 

Dependencies, such as Docker or Etcd, should also not be changed unless absolutely necessary, and also just to fix critical bugs (so, at most patch version changes, not new major nor minor versions).

**Comment by mbforbes** [comment]
@bgrant0607 thank you for the rewrite—much better. PTAL.

**Comment by mbforbes** [comment]
@k8s-bot please test this flaaaaaaake again

**Comment by zmerlynn** [comment]
@k8s-bot test this

**Comment by zmerlynn** [comment]
LGTM. I'll merge this when it's green so you can have a PR free day last day. :)

**Comment by zmerlynn** [comment]
Merging doc-only change before Max departs.

**Pull request merged**